components/camel-pulsar/src/main/java/org/apache/camel/component/pulsar/PulsarConsumer.java

```diff
@@ -76,4 +90,31 @@ protected void doResume() throws Exception {
         return strategy.create(endpoint);
     }
 
+    private Collection<ExecutorService> subscribeWithThreadPool(Collection<Consumer<byte[]>> consumers, PulsarEndpoint endpoint)
+            throws Exception {
+        int numThreads = endpoint.getPulsarConfiguration().getNumberOfConsumerThreads();
+        return consumers.stream().map(consumer -> {
+            ExecutorService executor = Executors.newFixedThreadPool(numThreads);
```

Comment:
Use Camel's executor service API for this, see getExecutorServiceManager on CamelContext